golang/pkg/rnr/tasks.go

```diff
@@ -11,16 +11,18 @@ const (
 )
 
 func taskSchedState(pbt *pb.Task) int {
-	m := map[pb.TaskState]int{
-		pb.TaskState_FAILED:        DONE,
-		pb.TaskState_PENDING:       PENDING,
-		pb.TaskState_RUNNING:       RUNNING,
-		pb.TaskState_SKIPPED:       DONE,
-		pb.TaskState_SUCCESS:       DONE,
-		pb.TaskState_ACTION_NEEDED: RUNNING,
+	switch pbt.State {
+	case pb.TaskState_FAILED, pb.TaskState_SKIPPED, pb.TaskState_SUCCESS:
+		return DONE
+
+	case pb.TaskState_PENDING:
+		return PENDING
+
+	case pb.TaskState_ACTION_NEEDED, pb.TaskState_RUNNING:
+		return RUNNING
 	}
 
-	return m[pbt.State]
+	return 0 // What should we do here?
```

Comment:
I'd probably panic (although it's not a great UX). We should probably extend the test from the previous commit to somehow identify all the valid values from the protobuf and ensure that taskSchedState is returning some value for them.

Reply:
That's a good idea. I coded a quick hack inside the test:

```go
	// test we defined results for all pb.TaskState
	for name, state := range pb.TaskState_value {
		t.Run(fmt.Sprintf("pb.TaskState_%s", name), func(t *testing.T) {
			got := taskSchedState(&pb.Task{State: pb.TaskState(state)})

			if name == "UNKNOWN" {
				if got != 0 {
					t.Errorf("UNKNOWN should be map to 0, got %v", got)
				}
			} else if got == 0 {
				t.Errorf("missing mapping for TaskState_%s", name)
			}
		})
	}
```

And the funniest thing happened: I ran the test and we are missing a mapping for `pb.TaskState_PENDING`.

Reply:
Wait, we **do** have a mapping for `PENDING`, which is indeed 0. What we don't have is one for unknown values, which should be 0. I'm going to add that.

Reply:
I've committed those changes, let me know what you think.

Reply:
Definitely LGTM. IIRC originally this PENDING was meant as a `default` scheduling state that doesn't do much harm, but adding an unknown state is definitely something that would make our life much easier.